Dockerfile

```diff
@@ -1,12 +1,41 @@
-from ruby:3.2.0
-
-env \
-  DEBIAN_FRONTEND=noninteractive \
-  NODE_VERSION=19.0.0
-
-run \
-  sed -i "/deb-src/d" /etc/apt/sources.list && \
-  apt-get update && \
-  apt-get install -y build-essential libpq-dev postgresql-client unzip cmake && \
-  curl -sSL "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.xz" | tar xfJ - -C /usr/local --strip-components=1 && \
-  npm install -g yarn
+from ruby:3.2.1-alpine
+
+RUN apk add postgresql-dev postgresql-client
+RUN apk add nodejs npm
+RUN apk add tzdata
+
+# for nokogiri
+RUN apk add --virtual .ruby-gemdeps libc-dev gcc libxml2-dev libxslt-dev make libffi-dev
+# for sassc
+RUN apk add g++
+
+RUN npm install -g yarn
+
+ENV TZ Pacific/Auckland
+ENV BUNDLE_JOBS 8
+
+ENV APP_HOME /myapp
+
+ARG RAILS_ENV=production
+ENV RAILS_ENV=$RAILS_ENV
+ENV RACK_ENV=$RAILS_ENV
+
+COPY . $APP_HOME
+WORKDIR $APP_HOME
+
+RUN if [ "$RAILS_ENV" = "production" ]; then \
+    bundle config set --local without 'development test' \
+    && bundle install \
+    && yarn install \
+    && bundle exec rake assets:precompile \
+    && rm -f /swapfile \
+    && dd if=/dev/zero of=/swapfile bs=1024 count=1048576 \
+    && mkswap /swapfile \
+    && chmod 600 /swapfile; \
+    fi
+
+RUN rm Dockerfile
```

Comment:
You don't need that. It is ok, and maybe even helpful to leave the `Dockerfile` there in the final image.

But anyway, you can add `Dockerfile` to the `.dockerignore` 😉 .